dev/tasks/r/github.nightly.yml

```diff
@@ -0,0 +1,381 @@
+          INSTALL_opts <- "--build"
+          if (!on_windows) {
+            # Windows doesn't support the --strip arg
+            INSTALL_opts <- c(INSTALL_opts, "--strip")
+          }
+
+          options(arrow.dev_repo = "https://nightly.wujciak.de/r")
+          install.packages(
+            "arrow",
+            type = "source",
+            repos = "{{ repo_domain }}",
+            INSTALL_opts = INSTALL_opts
+          )
+
+          # Test
+          library(arrow)
+          read_parquet(system.file("v0.7.1.parquet", package = "arrow"))
+      - name: Upload package
+        shell: bash
+        env:
+          USER: {{ '${{ secrets.CROSSBOW_NIGHTLIES_USER }}' }}
+          PW: {{ '${{ secrets.CROSSBOW_NIGHTLIES_TOKEN }}' }}
+          VERSION: {{ '${{ needs.source.outputs.version }}' }}
+        run: |
+          REPO_PATH=r$(Rscript -e "cat(contrib.url('', type = 'binary'))")
+          EXT=$(if [[ {{ '${{ matrix.platform }}' }} == windows* ]]; then echo zip; else echo tgz; fi)
+
+          curl -s --fail --show-error -u $USER:$PW -T arrow_$VERSION.$EXT {{ webdav_domain }}/$REPO_PATH/
+      - name: Upload binary artifact (temp)
+        uses: actions/upload-artifact@v3
+        with:
+          name: r-{{ '${{ matrix.config.os}}' }}-pkg
+          path: arrow_*.zip 
+
+  test-linux-binary:
+    if: true
+    needs: [source, linux-cpp]
+    name: Test binary {{ '${{ matrix.image }}' }}
+    runs-on: ubuntu-latest
+    container: {{ '${{ matrix.image }}' }}
+    strategy:
+      fail-fast: false
+      matrix:
+        image:
+          - "rhub/ubuntu-gcc-release" # ubuntu-20.04 (focal)
+          - "rstudio/r-base:4.1-bionic"
+          - "rstudio/r-base:4.2-centos7"
+          - "rocker/r-ver:3.6.3" # for debian:buster (10)
+          - "rocker/r-ver" # ubuntu-20.04
+          - "rhub/fedora-clang-devel" # tests distro-map.csv, mapped t
+    steps:
+      - name: Install system requirements
+        shell: bash
+        run: |
+          if [ "`which dnf`" ]; then
+            dnf install -y libcurl-devel openssl-devel
+          elif [ "`which yum`" ]; then
+            yum install -y libcurl-devel openssl-devel
+          elif [ "`which zypper`" ]; then
+            zypper install -y libcurl-devel libopenssl-devel
+          else
+            apt-get update
+            apt-get install -y libcurl4-openssl-dev libssl-dev
+          fi
+
+          # Add R-devel to PATH
+          echo "/opt/R-devel/bin" >> $GITHUB_PATH
+
+      - name: Set dev repo
+        shell: bash
+        run: |
+          echo 'options(arrow.dev_repo = "{{ repo_domain }}")' >> ~/.Rprofile
+      - name: Install arrow from our repo
+        env:
+          LIBARROW_BUILD: "FALSE"
+          LIBARROW_BINARY: "TRUE"
+        shell: Rscript {0}
+        run: |
+          {{  macros.test_r_src_pkg()|indent(8) }}
+
+  test-source:
+    #TODO Make sure we don't install arrow from CRAN if the repo fails

```

Comment:
If you install the dependencies from CRAN (or install.packages("arrow") from CRAN, then remove.packages("arrow") like we do elsewhere), then install.packages("arrow", repos = nightly_repo), there is no CRAN fallback.